src/Pact/Repl/Lib.hs

```diff
@@ -300,23 +302,39 @@ continuePact i as = case as of
   where
     go :: Integer -> Bool -> Maybe Text -> Maybe (ObjectMap (Term Name)) -> Eval LibState (Term Name)
     go step rollback pid userResume = do
-      pactExec <- use evalPactExec
-      pactId <- case pid of
-        Nothing -> maybe
-                   (evalError' i "continue-pact: no pact exec in context")
-                   (return . _pePactId)
-                   pactExec
-        Just pidTxt -> return $ PactId pidTxt
-      let pactStep = PactStep
-                     (fromIntegral step)
-                     rollback pactId userResume
+      pe <- use evalPactExec
+      (pactId, y) <- unwrapExec pid userResume pe
+
+      let pactStep = PactStep (fromIntegral step) rollback pactId y
       viewLibState (view rlsPacts) >>= \pacts ->
         case M.lookup pactId pacts of
           Nothing -> evalError' i $ "Invalid pact id: " <> pretty pactId
           Just PactExec{..} -> do
             evalPactExec .= Nothing
             local (set eePactStep $ Just pactStep) $ resumePact (_faInfo i) Nothing
 
+    unwrapExec mpid mobj Nothing = do
+      pid <- case mpid of
+        Nothing -> evalError' i
+          "continue-pact: No pact id supplied and no pact exec in context"
+        Just pid -> return $ PactId pid
+      y <- maybe (return Nothing) toYield mobj
+      return (pid, y)
+    unwrapExec mpid mobj (Just ex) = do
+      let pid = maybe (_pePactId ex) PactId mpid
+      y <- case mobj of
+        Nothing -> return $ _peYield ex
+        Just o -> case _peYield ex of
+          Nothing -> toYield o
+          Just (Yield _ Nothing) -> toYield o
+          Just (Yield _ (Just (Provenance tid _))) -> do
+            cid <- view $ eePublicData . pdPublicMeta . pmChainId
+            unless (cid == tid) $
+              evalError' i "resume overloads must occur on correct chain"
```

Comment:
We don't need to enforce this here either right? `resume` should catch this

Reply:
Yes - I missed a case in the `continue-pact` work. You'll notice i forgot to use the old provenance data from the `PactExec` and simply replace the object in the `Yield`. I've fixed it and tests pass!